### ISSUE TITLE
[Feature Request] Expand workspaces to include virtual monitors

### ISSUE THREAD

**Issue by LouChiSoft** [pop-os/cosmic-epoch#99]
Hi, first off I will just mention that I did previously ask about this but in the wrong repo (the GNOME version), I also mentioned it on Reddit [here](https://www.reddit.com/r/pop_os/comments/10qgosc/is_there_an_official_channel_for_feature_requests/) and it got some positive feedback but unfortunately nothing from a dev.

The idea is mainly for people who have larger (40+ inch) or ultra/super-wide monitors that would let them divide up a physical monitors space into smaller virtual monitors that behave and act like real physical monitors. I feel like it can definitely be a productivity boon as it means the user has more flexibility to set up their workspace, and depending on what they are doing at any given time they might want to use different layouts.
Also for ultra/super-wide monitors not all content scales well on the wider aspect ratio (especially a lot of older games) so being able to carve out a central 16:9 portion of the monitor that the game can run in and still have fully functional "side monitors" has the potential to unlock a whole new category of flexibility.

I would also like to point out that I am aware that cosmic-epoch is pre-alpha and I would not expect this to be a high priority feature. But I do think it would really make it stand out against other desktop environments

**Comment by LouChiSoft** [comment]
Hi, I spoke to @Drakulix in the Smithay Matrix chatroom and we discussed the feasability of such a feature and they said that it definitely could be done. They also explained you guys are very busy with the first alpha and already existing features and I understand that. I did want to say that I am would be willing to learn about Wayland and Smithay to be able to add this feature myself once I understand it well enough.

**Comment by joaomcarlos** [comment]
@LouChiSoft Wonder if you did get onboard with developing this?

**Comment by LouChiSoft** [comment]
Hi, I was wondering if it was possible to discuss this further. Alpha 1 and 2 of of the DE has released, so I wanted to know if there would be any interest in getting the ball rolling on this feature? Thanks

**Comment by joaomcarlos** [comment]
I'd love to have this. As an owner of an ultra wide monitor, this feature is basically critical for day to day use. 

Currently using half assed workarounds, which suck.

**Comment by dustymabe** [comment]
This is a critical feature. I'm currently [stuck](https://github.com/swaywm/sway/issues/1666#issuecomment-1356871296) on `i3` using the `fake-outputs` feature. As far as I know, no wayland based desktop supports this.

This feature would win over a lot of users. Is there anything at all we can do to help get this feature implemented?

**Comment by LouChiSoft** [comment]
I think the best we can do is just make people aware of the ticket? The more people who ask for it maybe it will incentivise System76 to prioritise it

**Comment by quack-o** [comment]
Indeed, i dont even have an ultrawide monitor, and i would love to have this as a feature, coming here from sway, its kinda embarrassing that no one is giving it any attention even tho sway is almost completed

**Comment by BeatScherrer** [comment]
this is exactly the reason I am still using herbstluftwm, it's done very well there. I have settings for 30/40/30, 25/50/25 and 20/60/20 splits which is all I ever need on my 32:9 monitor. I am currently testing cosmic again as my daily driver but with an ultra wide monitor this is currently the most missing feature compared to my herbstluftwm setup.

**Comment by Xpolo29** [comment]
Hey I would need this for streaming a virtual screen over Sunshine instead of a physical one !

**Comment by Solarunit** [comment]
I wonder about a cursor transition problem between such virtual monitors. Is it blocked?
For my usecase I would like to disable mouse cursor transition between 2 monitors. And maybe use some meta key pressed when I need to drag some window to it.
